dataRetriever/blockchain/baseBlockchain.go

```diff
@@ -70,3 +71,25 @@ func (bbc *baseBlockChain) SetCurrentBlockHeaderHash(hash []byte) {
 	bbc.currentBlockHeaderHash = hash
 	bbc.mut.Unlock()
 }
+
+func (bbc *baseBlockChain) SetHighestFinalBlockCoordinates(nonce uint64, headerHash []byte, rootHash []byte) {
+	bbc.mut.Lock()
+
+	bbc.finalCoordinates.blockNonce = nonce
+	bbc.finalCoordinates.blockHash = headerHash
+	bbc.finalCoordinates.blockRootHash = rootHash
```

Comment:
can we call this: committed root hash? instead of root hash?
on shard this will be the scheduled root hash, and on meta as we do not have yet scheduled txs support, it is the root hash

Reply:
Renamed as follows:

```
type blockInfo struct {
	nonce             uint64
	hash              []byte
	committedRootHash []byte
}
```